### ISSUE TITLE
Incorrect parsing of C enum comments defined using a macro

### ISSUE THREAD

**Issue by gilkalish** [doxygen/doxygen#6764]
I’m using macros to define enums (and strings) in C:

```
#define GENERATE_ENUM(ENUM, STR) ENUM,

typedef enum action {
    FOREACH_ACTION(GENERATE_ENUM)
} action_t;

#define FOREACH_ACTION(F)  \
    F(ACTION_1 = 1, "Action 1") \
    F(ACTION_2 = 2 /**< Some Doxygen comment */, "Action 2”) \
    F(ACTION_3 = 3 /**< Some other Doxygen comment */, "Action 3") \
    F(ACTION_MIN = ACTION_1, "") \
    F(ACTION_MAX = ACTION_3, "")
```

I’m using doxygen 1.8.14 with MACRO_EXPANSION=YES in Doxyfile (creating HTML output).
For some reason ACTION_3 is not being parsed correctly:
![capture](https://user-images.githubusercontent.com/46576896/50994009-e8ead280-14cf-11e9-9628-d09b1ba9a2a7.JPG)

**Comment by albert-github** [comment]
Can you please add your doxygen configuration file (Doxyfile) as well and also try with the current (1.8.15) version?

**Comment by gilkalish** [comment]
Same output for Doxygen 1.8.16.
Attaching my Doxyfile.
[DoxyfileSDK.zip](https://github.com/doxygen/doxygen/files/2756385/DoxyfileSDK.zip)

**Comment by albert-github** [comment]
I tried to produce the output as indicated but I didn't succeed (no enum output at all) with your Doxyfile. Though I got the message:
`XSOURCEX/aa.h:14: warning: documentation for unknown define FOREACH_ACTION found.`

Actions I took:
- created file aa,h with the content from the question and placing it in the directory XSOURCEX
- setting HTML_FOOTER = (I don't have the footer.html, but that is probably not the problem here).
- ran doxygen (1.8.15 and also the current master version, i.e. 1.8.16).

Can you please attach a, small, self contained example (source+config file in a tar or zip) that allows us to reproduce the problem? Also include the HTML output (For completeness also include the Doxyfile again)

**Comment by gilkalish** [comment]
Thanks Albert for your help,
I've managed to isolate the problem and it looks like it is only reproducing if I have a comma in the comment.
Attached is a zip file with the following:
- main.c, types.h, Makefile, generate_doxygen.sh - a small code files example
- customdoxygen.css, footer.html, header.html - all generic
[bug_doxygen.zip](https://github.com/doxygen/doxygen/files/2757313/bug_doxygen.zip)
.
- Doxyfile with my minimal changes
- html - a folder with the generated html code.

You can easily create the doxygen html output (and the binary) using the makefile by running "make doxygen_create"

Everything is using the latest master version (1.8.16), running on CentOS 7.4

In the html output if you look under the file types.h and the enum action you can see:
![capture1](https://user-images.githubusercontent.com/46576896/51145237-1d2ffd00-1808-11e9-9dc2-cdec8482e0b8.JPG)

But if I change the comment in the .h file from "Like 0, but cooler" to "Like 0. but cooler". It seems to work fine:
![capture2](https://user-images.githubusercontent.com/46576896/51145273-3b95f880-1808-11e9-86c5-665b3c5dd887.JPG)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6772

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
